script.js

```diff
@@ -47,37 +63,68 @@ function handleSearchInput(event) {
   renderBySearch();
 }
 
+const showSelector = document.getElementById("show-selector");
+showSelector.addEventListener("change", handleShowSelect);
+
+function handleShowSelect(event) {
+  state.selectedShowId = event.target.value;
+  console.log(state.selectedShowId);
+  fetchEpisodes();
+}
+
+function createShowListItem(show) {
+  const showListItem = document.getElementById("show-list").content.cloneNode(true);
+  const option = showListItem.querySelector("option");
+  option.textContent = show.name;
+  option.setAttribute("value", show.id);
+  return showListItem;
+}
+
 const episodeSelector = document.getElementById("episode-selector");
-episodeSelector.addEventListener("change", handleSelect);
+episodeSelector.addEventListener("change", handleEpisodeSelect);
 
-function handleSelect(event) {
+function handleEpisodeSelect(event) {
   state.selectedEpisodeId = event.target.value;
   console.log(state.selectedEpisodeId);
-  renderBySelect();
+  renderByEpisodeSelect();
 }
 
 function createEpisodeListItem(episode) {
-  const episodeListItem = document
-    .getElementById("episode-list")
-    .content.cloneNode(true);
+  const episodeListItem = document.getElementById("episode-list").content.cloneNode(true);
   const option = episodeListItem.querySelector("option");
-  option.textContent = `${episodeCode(episode.season, episode.number)} - ${
-    episode.name
-  }`;
+  option.textContent = `${episodeCode(episode.season, episode.number)} - ${episode.name}`;
   option.setAttribute("value", episode.id);
 
   return episodeListItem;
 }
 
+function fillShowList() {
+  const shows = state.shows;
+  for (s of shows) {
```

Comment:
I'd recommend using the full word in cases like this - if you use show it is a far easier to read at a glance than s

Reply:
done!